### ISSUE TITLE
Tag version regex matches non-dot characters

### ISSUE THREAD

**Issue by belcher-rok** [PaulHatch/semantic-version#78]
The regex at [this line](https://github.com/PaulHatch/semantic-version/blob/2fa33887a3b99ebd2c24d16c6c432c0464c1655d/src/formatting/DefaultTagFormatter.ts#L69) will match values like "1+2=3" due to unescaped dot literal (`.`) in regex expression.

Solution: Escape dot literals (`\.`) in all regex. 

Also consider escaping the `namespaceSeperator` and `namespace` values on line 66:
```typescript
function escape(literal: string) {
    return literal.replace(/\W/g, '\\$&'); // escape all non-word characters
}
const namespaceSeperator = escape(this.namespaceSeperator);
const namespace = escape(this.namespace);
return new RegExp(`^${this.tagPrefix}[0-9]+/.[0-9]+/.[0-9]+${namespaceSeperator}${namespace}$`).test(tag)
```

**Comment by PaulHatch** [comment]
Thanks, good catch.

**Comment by PaulHatch** [comment]
Fix in v5.0.1
